go/ql/test/experimental/CWE-522-DecompressionBombs/test.go

```diff
@@ -0,0 +1,301 @@
+				_ = rc.Close()
+				break
+			}
+			fmt.Print(result)
+			_ = rc.Close()
+		}
+	}
+}
+
+func ZipNewReader(file io.Reader) {
+	file1, _ := io.ReadAll(file)
+	zipReader, _ := zip.NewReader(bytes.NewReader(file1), int64(32<<20))
+	for _, file := range zipReader.File {
+		fileWriter := bytes.NewBuffer([]byte{})
+		fileReaderCloser, _ := file.Open()
+		result, _ := io.Copy(fileWriter, fileReaderCloser)
+		fmt.Print(result)
+	}
+}
+
+func ZipNewReader2(file io.Reader) {
+	file2, _ := io.ReadAll(file)
+	zipReaderKlauspost, _ := zipKlauspost.NewReader(bytes.NewReader(file2), int64(32<<20))
+	for _, file := range zipReaderKlauspost.File {
+		fileWriter := bytes.NewBuffer([]byte{})
+		// file.OpenRaw()
+		fileReaderCloser, _ := file.Open()
+		result, _ := io.Copy(fileWriter, fileReaderCloser)
+		fmt.Print(result)
+	}
+}
+func serve(w http.ResponseWriter, r *http.Request) {
+	if r.Body != nil {
+		if data, err := ioutil.ReadAll(r.Body); err == nil {
+			fmt.Println(data)
+		}
+	}
+}
+
+func TarDecompressor(file io.Reader, compressionType string) {
+	var tarRead *tar.Reader
+	if compressionType == "bzip2Dsnet" {
+		bzip2dsnet, _ := bzip2Dsnet.NewReader(file, &bzip2Dsnet.ReaderConfig{})
+		var out []byte = make([]byte, 70)
+		bzip2dsnet.Read(out)
+		tarRead = tar.NewReader(bzip2dsnet)
+	}
+	if compressionType == "bzip2" {
+		Bzip2 := bzip2.NewReader(file)
+		var out []byte = make([]byte, 70)
+		Bzip2.Read(out)
+		tarRead = tar.NewReader(Bzip2)
+	}
+	if compressionType == "flate" {
+		//flate.NewReaderDict()
+		Flate := flate.NewReader(file)
+		var out []byte = make([]byte, 70)
+		Flate.Read(out)
+		tarRead = tar.NewReader(Flate)
+	}
+	if compressionType == "flateKlauspost" {
+		//flateKlauspost.NewReaderDict()
+		zlibklauspost := flateKlauspost.NewReader(file)
+		var out []byte = make([]byte, 70)
+		zlibklauspost.Read(out)
+		tarRead = tar.NewReader(zlibklauspost)
+	}
+	if compressionType == "flateDsnet" {
+		flatedsnet, _ := flateDsnet.NewReader(file, &flateDsnet.ReaderConfig{})
+		var out []byte = make([]byte, 70)
+		flatedsnet.Read(out)
+		tarRead = tar.NewReader(flatedsnet)
+	}
+	if compressionType == "zlibKlauspost" {
+		//zlibKlauspost.NewReaderDict()
+		zlibklauspost, _ := zlibKlauspost.NewReader(file)
+		var out []byte = make([]byte, 70)
+		zlibklauspost.Read(out)
+		tarRead = tar.NewReader(zlibklauspost)
+	}
+	if compressionType == "zlib" {
+		//zlib.NewReaderDict()
+		Zlib, _ := zlib.NewReader(file)
+		var out []byte = make([]byte, 70)
+		Zlib.Read(out)
+		tarRead = tar.NewReader(Zlib)
+	}
+	if compressionType == "snappy" {
+		Snappy := snappy.NewReader(file)
+		var out []byte = make([]byte, 70)
+		Snappy.Read(out)
+		Snappy.ReadByte()
+		tarRead = tar.NewReader(Snappy)
+	}
+	if compressionType == "snappyKlauspost" {
+		snappyklauspost := snappyKlauspost.NewReader(file)
+		var out []byte = make([]byte, 70)
+		snappyklauspost.Read(out)
+		var buf bytes.Buffer
+		snappyklauspost.DecodeConcurrent(&buf, 2)
+		snappyklauspost.ReadByte()
```

Comment:
Is there meant to be a result here? There isn't one in the .expected file.

Reply:
In the stubs they are just two unrelated types. You can edit the stubs.

Reply:
Ah. Well, you should edit the tests anyway, and you can use them to debug the problem. From a CodeQL point of view, sometimes code that is dealing with types needs to use `getUnderlyingType` to look through aliases.